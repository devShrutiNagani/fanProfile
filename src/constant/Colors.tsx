export const COLORS = {
  theme: '#DD5E60',
  lightTheme: 'rgba(255, 204, 205, 0.3)',
  black: '#1D1D1D',
  white: '#FFFFFF',
  silver: '#E7E7E7',
  red: '#FF0000',
  grey: '#808080',
  lightGrey: '#EFEEEE',
  warmGrey: 'rgb(246, 243, 243)',
};
