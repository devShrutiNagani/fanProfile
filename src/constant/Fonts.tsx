export const FONTS: {[key in string]: string} = {
  montserratLight: 'Montserrat-Light',
  montserratRegular: 'Montserrat-Regular',
  montserratMedium: 'Montserrat-Medium',
  montserratSemiBold: 'Montserrat-SemiBold',
  montserratBold: 'Montserrat-Bold',
};
