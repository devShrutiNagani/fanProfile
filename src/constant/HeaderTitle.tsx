export const USER_DATA = 'User Data';
export const DRIVER_DATA = 'Driver Data';
export const DRIVER_DETAILS = 'Driver Details';
