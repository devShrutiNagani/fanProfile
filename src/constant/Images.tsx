export const ICONS = {
  email: require('../assets/icons/email.png'),
  name: require('../assets/icons/user.png'),
  backIcon: require('../assets/icons/backIcon.png'),
  link: require('../assets/icons/link.png'),
};

export const IMAGES = {
  noData: require('../assets/images/noData.png'),
  driver: require('../assets/images/driver.jpg'),
  userDataImg: require('../assets/images/userDataImg.jpg'),
};
