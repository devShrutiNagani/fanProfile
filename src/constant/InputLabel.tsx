export const NAME = 'Name';
export const EMAIL = 'Email';
