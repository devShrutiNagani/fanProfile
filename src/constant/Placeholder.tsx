export const ENTER_NAME = 'Enter Name';
export const ENTER_EMAIL = 'Enter Email';
