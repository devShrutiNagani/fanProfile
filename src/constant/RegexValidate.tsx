export const RegexValidate = {
  emailRegex: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
};
