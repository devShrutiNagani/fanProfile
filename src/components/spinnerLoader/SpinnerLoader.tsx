import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import createStyles from './Styles';
import {SpinnerLoaderProps} from '../../constant/interface/custom/Custom';
import {COLORS} from '../../constant/Colors';

const SpinnerLoader: React.FC<SpinnerLoaderProps> = ({
  size = 'large',
  color,
  style,
}) => {
  const styles = createStyles();
  return (
    <View style={[styles.container, style]}>
      <ActivityIndicator
        size={size}
        color={color || COLORS.theme}
        testID="spinner-loader"
      />
    </View>
  );
};

export default SpinnerLoader;
