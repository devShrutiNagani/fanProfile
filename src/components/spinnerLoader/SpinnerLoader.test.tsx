import React from 'react';
import {render} from '@testing-library/react-native';
import {COLORS} from '../../constant/Colors';
import SpinnerLoader from './SpinnerLoader';

describe('SpinnerLoader Component', () => {
  it('renders ActivityIndicator with default size and color', () => {
    const {getByTestId} = render(<SpinnerLoader />);

    const spinner = getByTestId('spinner-loader');
    expect(spinner).toBeTruthy();
    expect(spinner.props.size).toBe('large'); // Default size
    expect(spinner.props.color).toBe(COLORS.theme); // Default color
  });

  it('renders ActivityIndicator with custom size and color', () => {
    const {getByTestId} = render(<SpinnerLoader size="small" color="red" />);

    const spinner = getByTestId('spinner-loader');
    expect(spinner.props.size).toBe('small'); // Custom size
    expect(spinner.props.color).toBe('red'); // Custom color
  });

  it('matches snapshot', () => {
    const tree = render(<SpinnerLoader size="small" color="blue" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
