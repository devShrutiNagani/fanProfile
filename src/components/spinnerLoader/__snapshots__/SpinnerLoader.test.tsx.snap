// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerLoader Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    color="blue"
    size="small"
    testID="spinner-loader"
  />
</View>
`;
